### ISSUE TITLE
Replace ALB annotations with IngressClassParams

### ISSUE THREAD

**Issue by stevehipwell** [kubernetes-sigs/aws-load-balancer-controller#2311]
**Is your feature request related to a problem?**
I'd like to be able to replace more of the `alb.ingress.kubernetes.io` annotations with the `IngressClassParams` to abstract away the ALB specification from the `Ingress` resources using it. I mistakenly thought that #2190 was doing this, but after re-reading I think that's just replacing the `alb.ingress.kubernetes.io/load-balancer-attributes` annotation.

**Describe the solution you'd like**
I'd like to be able to specify the following annotations in the `IngressClassParams` directly.

- alb.ingress.kubernetes.io/target-type
- alb.ingress.kubernetes.io/subnets
- alb.ingress.kubernetes.io/listen-ports
- alb.ingress.kubernetes.io/ssl-redirect
- alb.ingress.kubernetes.io/inbound-cidrs
- alb.ingress.kubernetes.io/certificate-arn

**Describe alternatives you've considered**
n/a

**Comment by nikskiz** [comment]
It would be nice to have `spec.params.targetGroupAttributes`

**Comment by Nuru** [comment]
Really, I think any of the 14 annotations that are "Exclusive" (must only be set once per Group) should be configurable via IngressClass (possibly via IngressClassParams). Then we can move that configuration out of the individual services so we do not have to worry about them fighting over control of them. 

As of version 2.4, the following annotations are marked "Exclusive", but only a few of them can be specified in IngressClass and most (all?) of the rest cannot be specified anywhere but via annotations. This is a poor separation of concerns.

1. [alb.ingress.kubernetes.io/load-balancer-name](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#load-balancer-name)
2. [alb.ingress.kubernetes.io/ip-address-type](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#ip-address-type)
3. [alb.ingress.kubernetes.io/scheme](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#scheme)
4. [alb.ingress.kubernetes.io/subnets](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#subnets)
5. [alb.ingress.kubernetes.io/security-groups](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#security-groups)
6. [alb.ingress.kubernetes.io/manage-backend-security-group-rules](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#manage-backend-security-group-rules)
7. [alb.ingress.kubernetes.io/customer-owned-ipv4-pool](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#customer-owned-ipv4-pool)
8. [alb.ingress.kubernetes.io/load-balancer-attributes](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#load-balancer-attributes)
9. [alb.ingress.kubernetes.io/wafv2-acl-arn](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#wafv2-acl-arn)
10. [alb.ingress.kubernetes.io/waf-acl-id](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#waf-acl-id)
11. [alb.ingress.kubernetes.io/shield-advanced-protection](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#shield-advanced-protection)
12. [alb.ingress.kubernetes.io/ssl-redirect](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#ssl-redirect)
13. [alb.ingress.kubernetes.io/inbound-cidrs](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#inbound-cidrs)
14. [alb.ingress.kubernetes.io/ssl-policy](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#ssl-policy)

Related to, but different than #2600

**Comment by kwohlfahrt** [comment]
I would be particularly interested in the [alb.ingress.kubernetes.io/auth-*](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#auth-type) being defined with class params.

That would make it easy to set up an ingress class with sensible authentication defaults for external ingresses.

**Comment by johngmyers** [comment]
#2920 is chipping away the ones that have `Exclusive` `MergeBehavior`. For the ones that have `Merge`, `listen-ports` has per-Ingress semantics that complicate things. `certificate-arn` has good reasons for specifying per-Ingress.

**Comment by visit1985** [comment]
> `certificate-arn` has good reasons for specifying per-Ingress.

Agreed, but having a default cert in Ingress ClassParams would still make sense.

**Comment by lgg42** [comment]
> Really, I think any of the 14 annotations that are "Exclusive" (must only be set once per Group) should be configurable via IngressClass (possibly via IngressClassParams). Then we can move that configuration out of the individual services so we do not have to worry about them fighting over control of them.
> 
> As of version 2.4, the following annotations are marked "Exclusive", but only a few of them can be specified in IngressClass and most (all?) of the rest cannot be specified anywhere but via annotations. This is a poor separation of concerns.
> 
> 1. [alb.ingress.kubernetes.io/load-balancer-name](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#load-balancer-name)
> 2. [alb.ingress.kubernetes.io/ip-address-type](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#ip-address-type)
> 3. [alb.ingress.kubernetes.io/scheme](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#scheme)
> 4. [alb.ingress.kubernetes.io/subnets](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#subnets)
> 5. [alb.ingress.kubernetes.io/security-groups](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#security-groups)
> 6. [alb.ingress.kubernetes.io/manage-backend-security-group-rules](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#manage-backend-security-group-rules)
> 7. [alb.ingress.kubernetes.io/customer-owned-ipv4-pool](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#customer-owned-ipv4-pool)
> 8. [alb.ingress.kubernetes.io/load-balancer-attributes](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#load-balancer-attributes)
> 9. [alb.ingress.kubernetes.io/wafv2-acl-arn](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#wafv2-acl-arn)
> 10. [alb.ingress.kubernetes.io/waf-acl-id](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#waf-acl-id)
> 11. [alb.ingress.kubernetes.io/shield-advanced-protection](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#shield-advanced-protection)
> 12. [alb.ingress.kubernetes.io/ssl-redirect](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#ssl-redirect)
> 13. [alb.ingress.kubernetes.io/inbound-cidrs](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#inbound-cidrs)
> 14. [alb.ingress.kubernetes.io/ssl-policy](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/ingress/annotations/#ssl-policy)
> 
> Related to, but different than [#2600](https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/2600)

This makes so much sense!!!

**Comment by divisionium** [comment]
I definitely agree that at least load-balancer-name should be in the IngressClassParams spec.
